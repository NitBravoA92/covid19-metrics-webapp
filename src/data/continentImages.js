import Africa from '../assets/images/Africa.png';
import Asia from '../assets/images/Asia.png';
import AustraliaOceania from '../assets/images/AustraliaOceania.png';
import Europe from '../assets/images/Europe.png';
import NorthAmerica from '../assets/images/NorthAmerica.png';
import SouthAmerica from '../assets/images/SouthAmerica.webp';
import NewContinentFake from '../assets/images/NewContinentFake.png';

const continentImages = {
  Africa,
  Asia,
  'Australia-Oceania': AustraliaOceania,
  Europe,
  'North America': NorthAmerica,
  'South America': SouthAmerica,
  'New Continent Fake': NewContinentFake,
};

export default continentImages;
