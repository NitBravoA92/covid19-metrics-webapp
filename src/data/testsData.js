export const fakeContinentData = [
  {
    updated: 1234560000000,
    cases: 563549586,
    todayCases: 100,
    deaths: 1061592,
    todayDeaths: 3,
    recovered: 200000000,
    todayRecovered: 500,
    active: 1529370,
    critical: 6000,
    casesPerOneMillion: 212686.51,
    deathsPerOneMillion: 2740.67,
    tests: 2507653000,
    testsPerOneMillion: 2206050.64,
    population: 598140916,
    continent: 'North America',
    activePerOneMillion: 4016.25,
    recoveredPerOneMillion: 205929.59,
    criticalPerOneMillion: 9.79,
    continentInfo: {
      lat: 31.6768272,
      long: -146.4707474,
    },
    countries: [],
  },
  {
    updated: 7890123456000,
    cases: 540598247,
    todayCases: 2450,
    deaths: 9128563,
    todayDeaths: 150,
    recovered: 395740123,
    todayRecovered: 503,
    active: 40234759,
    critical: 20991,
    casesPerOneMillion: 46541.57,
    deathsPerOneMillion: 328.63,
    tests: 10842394059,
    testsPerOneMillion: 508887.44,
    population: 4711356783,
    continent: 'Asia',
    activePerOneMillion: 3230.04,
    recoveredPerOneMillion: 42982.89,
    criticalPerOneMillion: 3.21,
    continentInfo: {
      lat: 23.7027273,
      long: 62.3750637,
    },
    countries: [],
  },
  {
    updated: 9486574635637,
    cases: 185757446,
    todayCases: 3094,
    deaths: 3156000,
    todayDeaths: 200,
    recovered: 245885503,
    todayRecovered: 5400,
    active: 3814470,
    critical: 7000,
    casesPerOneMillion: 334063.2,
    deathsPerOneMillion: 2766.84,
    tests: 8923467182,
    testsPerOneMillion: 3791654.17,
    population: 747543837,
    continent: 'Europe',
    activePerOneMillion: 2371.84,
    recoveredPerOneMillion: 328924.53,
    criticalPerOneMillion: 7.32,
    continentInfo: {
      lat: 25.771324,
      long: -35.6012256,
    },
    countries: [],
  },
  {
    updated: 1690921777990,
    cases: 68891759,
    todayCases: 0,
    deaths: 1358906,
    todayDeaths: 0,
    recovered: 66504284,
    todayRecovered: 0,
    active: 1028569,
    critical: 10103,
    casesPerOneMillion: 157396.92,
    deathsPerOneMillion: 3104.69,
    tests: 244655471,
    testsPerOneMillion: 558964.08,
    population: 437694443,
    continent: 'South America',
    activePerOneMillion: 2349.97,
    recoveredPerOneMillion: 151942.26,
    criticalPerOneMillion: 23.08,
    continentInfo: {
      lat: -15.6551563,
      long: -100.7484231,
    },
    countries: [],
  },
  {
    updated: 1690921777991,
    cases: 14586751,
    todayCases: 0,
    deaths: 29807,
    todayDeaths: 0,
    recovered: 14444237,
    todayRecovered: 0,
    active: 112707,
    critical: 50,
    casesPerOneMillion: 335555.88,
    deathsPerOneMillion: 685.68,
    tests: 91667029,
    testsPerOneMillion: 2108722.54,
    population: 43470408,
    continent: 'Australia-Oceania',
    activePerOneMillion: 2592.73,
    recoveredPerOneMillion: 332277.47,
    criticalPerOneMillion: 1.15,
    continentInfo: {
      lat: -8.6599161,
      long: 91.1469847,
    },
    countries: [],
  },
  {
    updated: 1690921777992,
    cases: 12836207,
    todayCases: 0,
    deaths: 258822,
    todayDeaths: 0,
    recovered: 12087976,
    todayRecovered: 0,
    active: 489409,
    critical: 547,
    casesPerOneMillion: 9124.86,
    deathsPerOneMillion: 183.99,
    tests: 111020621,
    testsPerOneMillion: 78921.13,
    population: 1406728744,
    continent: 'Africa',
    activePerOneMillion: 347.91,
    recoveredPerOneMillion: 8592.97,
    criticalPerOneMillion: 0.39,
    continentInfo: {
      lat: 1.7383867,
      long: -16.3094636,
    },
    countries: [],
  },
  {
    updated: 9746354590,
    cases: 5000000,
    todayCases: 0,
    deaths: 2000,
    todayDeaths: 0,
    recovered: 4400000,
    todayRecovered: 0,
    active: 100000,
    critical: 100,
    casesPerOneMillion: 9124.86,
    deathsPerOneMillion: 183.99,
    tests: 20394857,
    testsPerOneMillion: 78921.13,
    population: 30874590,
    continent: 'New Continent Fake',
    activePerOneMillion: 347.91,
    recoveredPerOneMillion: 8592.97,
    criticalPerOneMillion: 0.39,
    continentInfo: {
      lat: 1.7383867,
      long: -16.3094636,
    },
    countries: [],
  },
];

export const expectedContinentData = [
  {
    id: 1,
    name: 'North America',
    population: '598,140,916',
    cases: '563,549,586',
    deaths: '1,061,592',
    recovered: '200,000,000',
    active: '1,529,370',
  },
  {
    id: 2,
    name: 'Asia',
    population: '4,711,356,783',
    cases: '540,598,247',
    deaths: '9,128,563',
    recovered: '395,740,123',
    active: '40,234,759',
  },
  {
    id: 3,
    name: 'Europe',
    population: '747,543,837',
    cases: '185,757,446',
    deaths: '3,156,000',
    recovered: '245,885,503',
    active: '3,814,470',
  },
  {
    id: 4,
    name: 'South America',
    population: '437,694,443',
    cases: '68,891,759',
    deaths: '1,358,906',
    recovered: '66,504,284',
    active: '1,028,569',
  },
  {
    id: 5,
    name: 'Australia-Oceania',
    population: '43,470,408',
    cases: '14,586,751',
    deaths: '29,807',
    recovered: '14,444,237',
    active: '112,707',
  },
  {
    id: 6,
    name: 'Africa',
    population: '1,406,728,744',
    cases: '12,836,207',
    deaths: '258,822',
    recovered: '12,087,976',
    active: '489,409',
  },
  {
    id: 7,
    name: 'New Continent Fake',
    population: '30,874,590',
    cases: '5,000,000',
    deaths: '2,000',
    recovered: '4,400,000',
    active: '100,000',
  },
];

export const fakeCountriesData = [
  {
    updated: 1691125210455,
    country: 'Afghanistan',
    countryInfo: {
      _id: 4,
      iso2: 'AF',
      iso3: 'AFG',
      lat: 33,
      long: 65,
      flag: 'https://disease.sh/assets/img/flags/af.png',
    },
    cases: 224317,
    todayCases: 0,
    deaths: 7940,
    todayDeaths: 0,
    recovered: 204341,
    todayRecovered: 0,
    active: 12036,
    critical: 45,
    casesPerOneMillion: 5504,
    deathsPerOneMillion: 195,
    tests: 1285637,
    testsPerOneMillion: 31546,
    population: 40754388,
    continent: 'Asia',
    oneCasePerPeople: 182,
    oneDeathPerPeople: 5133,
    oneTestPerPeople: 32,
    activePerOneMillion: 295.33,
    recoveredPerOneMillion: 5013.96,
    criticalPerOneMillion: 1.1,
  },
  {
    updated: 1691125210443,
    country: 'Albania',
    countryInfo: {
      _id: 8,
      iso2: 'AL',
      iso3: 'ALB',
      lat: 41,
      long: 20,
      flag: 'https://disease.sh/assets/img/flags/al.png',
    },
    cases: 334726,
    todayCases: 0,
    deaths: 3602,
    todayDeaths: 0,
    recovered: 329428,
    todayRecovered: 0,
    active: 1696,
    critical: 0,
    casesPerOneMillion: 116777,
    deathsPerOneMillion: 1257,
    tests: 1941032,
    testsPerOneMillion: 677173,
    population: 2866374,
    continent: 'Europe',
    oneCasePerPeople: 9,
    oneDeathPerPeople: 796,
    oneTestPerPeople: 1,
    activePerOneMillion: 591.69,
    recoveredPerOneMillion: 114928.48,
    criticalPerOneMillion: 0,
  },
  {
    updated: 1691125210448,
    country: 'Algeria',
    countryInfo: {
      _id: 12,
      iso2: 'DZ',
      iso3: 'DZA',
      lat: 28,
      long: 3,
      flag: 'https://disease.sh/assets/img/flags/dz.png',
    },
    cases: 271852,
    todayCases: 0,
    deaths: 6881,
    todayDeaths: 0,
    recovered: 183061,
    todayRecovered: 0,
    active: 81910,
    critical: 0,
    casesPerOneMillion: 5995,
    deathsPerOneMillion: 152,
    tests: 230960,
    testsPerOneMillion: 5093,
    population: 45350148,
    continent: 'Africa',
    oneCasePerPeople: 167,
    oneDeathPerPeople: 6591,
    oneTestPerPeople: 196,
    activePerOneMillion: 1806.17,
    recoveredPerOneMillion: 4036.61,
    criticalPerOneMillion: 0,
  },
];

export const expectedCountriesData = [
  {
    id: 1,
    name: 'Afghanistan',
    continent: 'Asia',
    flag: 'https://disease.sh/assets/img/flags/af.png',
    population: '40,754,388',
    cases: '224,317',
    todayCases: '0',
    deaths: '7,940',
    todayDeaths: '0',
    recovered: '204,341',
    todayRecovered: '0',
    active: '12,036',
    critical: '45',
    tests: '1,285,637',
  },
  {
    id: 2,
    name: 'Albania',
    continent: 'Europe',
    flag: 'https://disease.sh/assets/img/flags/al.png',
    population: '2,866,374',
    cases: '334,726',
    todayCases: '0',
    deaths: '3,602',
    todayDeaths: '0',
    recovered: '329,428',
    todayRecovered: '0',
    active: '1,696',
    critical: '0',
    tests: '1,941,032',
  },
  {
    id: 3,
    name: 'Algeria',
    continent: 'Africa',
    flag: 'https://disease.sh/assets/img/flags/dz.png',
    population: '45,350,148',
    cases: '271,852',
    todayCases: '0',
    deaths: '6,881',
    todayDeaths: '0',
    recovered: '183,061',
    todayRecovered: '0',
    active: '81,910',
    critical: '0',
    tests: '230,960',
  },
];
