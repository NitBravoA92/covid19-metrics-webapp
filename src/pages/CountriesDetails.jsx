import { useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import { BsChevronLeft } from 'react-icons/bs';
import continentImages from '../data/continentImages';
import Header from '../components/Header';
import Headline from '../components/Headline';
import SectionTitle from '../components/SectionTitle';
import CountryItem from '../components/CountryItem';
import '../assets/css/countries.css';

const CountriesDetails = () => {
  const { continentName } = useParams();

  const realContinentName = continentName === 'Australia-Oceania'
    ? continentName
    : continentName.replace(/-/g, ' ');

  const { continents } = useSelector((state) => state.continents);
  const { countries } = useSelector((state) => state.countries);

  const selectedContinent = continents.find((continent) => continent.name === realContinentName);

  const filteredCountries = countries.filter((country) => country.continent === realContinentName);

  return (
    <>
      <Header
        navigation={{ name: <BsChevronLeft />, url: '/', page: 'Continent stats' }}
      />
      <main>
        <Headline
          srcImg={continentImages[realContinentName]}
          name={realContinentName}
          information={[
            { stats: selectedContinent?.population, text: 'inhabitants' },
            { stats: selectedContinent?.cases, text: 'total cases' },
            { stats: selectedContinent?.deaths, text: 'deaths' },
            { stats: selectedContinent?.recovered, text: 'recovered' },
            { stats: selectedContinent?.active, text: 'active cases' },
          ]}
        />

        <section className="countries">
          <SectionTitle title="Stats by countries" />
          <div className="countries-list">
            {filteredCountries.map(({
              id, flag, name, cases,
            }) => (
              <CountryItem key={id} image={flag} name={name} cases={cases} />
            ))}
          </div>
        </section>
      </main>
    </>
  );
};

export default CountriesDetails;
