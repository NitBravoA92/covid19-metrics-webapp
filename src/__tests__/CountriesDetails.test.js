import { Routes, Route, MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { render, screen } from '@testing-library/react';
import continentsReducer, {
  getAllContinents,
} from '../redux/continents/continentsSlice';
import { fakeContinentData, fakeCountriesData } from '../data/testsData';
import countriesReducer, { getAllCountries } from '../redux/countries/countriesSlice';
import CountriesDetails from '../pages/CountriesDetails';

describe('The CountriesDetails page component', () => {
  let store;

  beforeEach(async () => {
    const continentState = continentsReducer(undefined, {
      type: getAllContinents.fulfilled,
      payload: fakeContinentData,
    });
    const countriesState = countriesReducer(undefined, {
      type: getAllCountries.fulfilled,
      payload: fakeCountriesData,
    });
    const thunkMiddleware = [thunk];
    const mockStore = configureStore(thunkMiddleware);
    const initialStateContinents = continentState;
    const initialState = countriesState;
    store = mockStore({
      continents: initialStateContinents,
      countries: initialState,
    });
    store.dispatch(getAllContinents());
    store.dispatch(getAllCountries());
  });

  test('should render correctly into the DOM', async () => {
    const { countriesDetailsComponent } = render(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/Africa']}>
          <Routes>
            <Route path=":continentName" element={<CountriesDetails />} />
          </Routes>
        </MemoryRouter>
      </Provider>,
    );
    expect(countriesDetailsComponent).toMatchSnapshot();
  });

  test('should render the second page of the app with the fake data of countries of Asia continent: Afghanistan', async () => {
    render(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/Asia']}>
          <Routes>
            <Route path=":continentName" element={<CountriesDetails />} />
          </Routes>
        </MemoryRouter>
      </Provider>,
    );

    const countriesItems = document.querySelectorAll('.country-item');
    const continentNameAsia = screen.getByText('Asia');
    const countryName = screen.getByText('Afghanistan');

    expect(countriesItems).toHaveLength(1);
    expect(continentNameAsia).toBeInTheDocument();
    expect(countryName).toBeInTheDocument();
  });
});
