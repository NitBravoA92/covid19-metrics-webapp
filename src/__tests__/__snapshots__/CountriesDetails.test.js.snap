// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The CountriesDetails page component should render correctly into the DOM 1`] = `undefined`;
